### PR TITLE
support /watch/, /v/ and /w/ subpaths

### LINKED ISSUE #534: Support youtube.com/v/videoId

### PR DESCRIPTION
**Pull request by B0pol**
fixes #534 fixes https://github.com/TeamNewPipe/NewPipe/issues/4283

- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [ ] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [ ] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

I trust the tests

### CODE REVIEW

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java [hunk 1]
```diff
@@ -237,4 +221,14 @@ public boolean onAcceptUrl(final String url) throws FoundAdException {
             return false;
         }
     }
+
+    private String getIdFromFoldersInPath(String path) throws ParsingException {
+        for (final String folder : folders) {
+            if (path.startsWith(folder)) {
+                String id = path.split("/")[1];
```
Comment (TobiGr) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java:228]: `substring` is way faster than `split`, so please use that method instead.

**Review by TobiGr — COMMENTED** [review]
looks good. two small things.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java [hunk 2]
```diff
@@ -38,6 +40,7 @@
 
     private static final Pattern YOUTUBE_VIDEO_ID_REGEX_PATTERN = Pattern.compile("^([a-zA-Z0-9_-]{11})");
     private static final YoutubeStreamLinkHandlerFactory instance = new YoutubeStreamLinkHandlerFactory();
+    private static final List<String> folders = Arrays.asList("embed/", "shorts/", "watch/", "v/", "w/");
```
[the remark this answers is not in the document]
Comment (B0pol) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java:43]: path containing what? folders

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java [hunk 3]
```diff
@@ -38,6 +40,7 @@
 
     private static final Pattern YOUTUBE_VIDEO_ID_REGEX_PATTERN = Pattern.compile("^([a-zA-Z0-9_-]{11})");
     private static final YoutubeStreamLinkHandlerFactory instance = new YoutubeStreamLinkHandlerFactory();
+    private static final List<String> subpaths = Arrays.asList("embed/", "shorts/", "watch/", "v/", "w/");
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java:43]:
```suggestion
    private static final List<String> SUBPATHS = Arrays.asList("embed/", "shorts/", "watch/", "v/", "w/");
```

**Review by Stypox — CHANGES_REQUESTED** [review]
LGTM

**Review by Stypox — APPROVED** [review]

**Pull request merged**